processing/src/main/java/io/druid/query/groupby/strategy/GroupByStrategyV2.java

```diff
@@ -141,9 +140,9 @@ public GroupByQueryResource prepareResource(GroupByQuery query, boolean willMerg
       } else if (requiredMergeBufferNum == 0) {
         return new GroupByQueryResource();
       } else {
-        final Number timeout = query.getContextValue(QueryContextKeys.TIMEOUT, JodaUtils.MAX_INSTANT);
+        final long timeout = QueryContexts.getTimeout(query);
         final ResourceHolder<List<ByteBuffer>> mergeBufferHolders = mergeBufferPool.takeBatch(
-            requiredMergeBufferNum, timeout.longValue()
+            requiredMergeBufferNum, timeout
```

Comment:
`takeBatch()` javadoc says that "negative means no timeout", here timeout is QueryContext.DEFAULT_TIMEOUT, that is 0.